### PR TITLE
Hw login3

### PR DESCRIPTION
**Pull request by Martynice**

### CODE REVIEW

**Review by tarkaiev — APPROVED** [review]

File: src/main/java/com/internet/shop/controllers/user/LoginController.java [hunk 1]
```diff
@@ -18,11 +19,16 @@
...
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
-        req.getRequestDispatcher("/WEB-INF/views/login.jsp").forward(req, resp);
+        if (userService.findByLogin("admin").isPresent()) {
+            req.getRequestDispatcher("/WEB-INF/views/login.jsp").forward(req, resp);
+        } else {
+            resp.sendRedirect(req.getContextPath() + "/inject");
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/user/LoginController.java]: let's better not change this and inject admin with a button

File: src/main/java/com/internet/shop/model/Role.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.internet.shop.model;
+
+public class Role {
+    private Long id;
+    private RoleName roleName;
+
+    public Role(RoleName roleName) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/Role.java]:
```suggestion
    private Role(RoleName roleName) {
```

File: src/main/webapp/WEB-INF/views/accessDenied.jsp [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Title</title>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/accessDenied.jsp]: add informative title

File: src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebFilter("/*")
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java]: can you guarantee the order in which your filters execute? let's indicate order in web.xml

File: src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            req.getRequestDispatcher("/WEB-INF/views/accessDenied.jsp").forward(req, resp);
+            return;
+        }
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    private boolean isAuthorizes(User user, List<Role.RoleName> authorizedRoles) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java]:
```suggestion
    private boolean isAuthorized(User user, List<Role.RoleName> authorizedRoles) {
```

File: src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        protectedUrl.put("/shopping-cart/product/delete", List.of(Role.RoleName.USER));
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String requestedUrl = req.getServletPath();
+        if (protectedUrl.get(requestedUrl) == null) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java]: we can also use `containsKey()` here

File: src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        protectedUrl.put("/users/all", List.of(Role.RoleName.ADMIN));
+        protectedUrl.put("/users/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrl.put("/order/all", List.of(Role.RoleName.ADMIN));
+        protectedUrl.put("/product/all-admin", List.of(Role.RoleName.ADMIN));
+        protectedUrl.put("/product/add", List.of(Role.RoleName.ADMIN));
+        protectedUrl.put("/product/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrl.put("/product/buy", List.of(Role.RoleName.USER));
+        protectedUrl.put("/order/complete", List.of(Role.RoleName.USER));
+        protectedUrl.put("/order/current-user", List.of(Role.RoleName.USER));
+        protectedUrl.put("/order/delete", List.of(Role.RoleName.USER));
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java]:
```suggestion
        protectedUrl.put("/order/delete", List.of(Role.RoleName.ADMIN));
```
only admin can delete orders

File: src/main/java/com/internet/shop/controllers/order/DeleteOrderController.java [hunk 8]
```diff
@@ -19,6 +19,6 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws IOException {
         Long orderId = Long.valueOf(req.getParameter("id"));
         orderService.delete(orderId);
-        resp.sendRedirect(req.getContextPath() + "/order/all");
+        resp.sendRedirect(req.getContextPath() + "/order/current-user");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/DeleteOrderController.java]:
```suggestion
        resp.sendRedirect(req.getContextPath() + "/order/all");
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controllers/order/DeleteOrderController.java (hunk 8 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/controllers/order/DeleteOrderController.java]: changed

File: src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java (hunk 7 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java]: changed

File: src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java (hunk 6 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java]: done

File: src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java (hunk 5 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java]: corrected

File: src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java (hunk 4 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java]: added

File: src/main/webapp/WEB-INF/views/accessDenied.jsp (hunk 3 above)
Comment (Martynice) [reply src/main/webapp/WEB-INF/views/accessDenied.jsp]: done

File: src/main/java/com/internet/shop/model/Role.java (hunk 2 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/model/Role.java]: changed

File: src/main/java/com/internet/shop/controllers/user/LoginController.java (hunk 1 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/controllers/user/LoginController.java]: added

File: src/main/java/com/internet/shop/controllers/user/LoginController.java [hunk 9]
```diff
@@ -18,6 +19,7 @@
     private static final Injector injector = Injector.getInstance("com.internet.shop");
     private final AuthenticationService authService =
             (AuthenticationService) injector.getInstance(AuthenticationService.class);
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/user/LoginController.java]: I think we don't need this anymore

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/internet/shop/controllers/user/LoginController.java (hunk 9 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/controllers/user/LoginController.java]: right

**Pull request merged**